File: src/ceo/cloudnation/cloudessentials/commands/HealCMD.java
package ceo.cloudnation.cloudessentials.commands;

import net.md_5.bungee.api.ChatColor;     
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import ceo.cloudnation.cloudessentials.Main;

public class HealCMD implements CommandExecutor {
	
	@Override
	public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {
		Player player = (Player) sender;
		if(label.equalsIgnoreCase("heal")) {
			if(args.length == 0) {
				player.sendMessage("§7You have been healed!");
				player.setHealth(20.0);
				player.setFoodLevel(20);
				player.setFireTicks(0);
			}
		}
		return false;
	}
}

File: src/main/kotlin/com/github/avexiie/commands/features/HealCMD.kt
package com.github.avexiie.commands.features

import com.github.avexiie.extensions.checkPermission
import com.github.avexiie.objects.Config
import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class HealCMD : CommandExecutor {

    override fun onCommand(
        sender: CommandSender,
        command: Command,
        commandName: String,
        args: Array<out String>
    ): Boolean {
        if (!sender.checkPermission("heal"))
            return false

        var target: Player

        // casts target
        target =
                // if console executes this
            if (sender !is Player) {
                // console must specify a player
                if (args.isEmpty()) {
                    sender.sendMessage(Config.prefix + " §cInvalid usage!")
                    return false
                }

                if (Bukkit.getPlayer(args[0]) == null) {
                    sender.sendMessage(Config.prefix + " §cPlayer not found!")
                    return false
                }

                Bukkit.getPlayer(args[0])

                // if executed by player
            } else
                sender

        var others = false

        if (args.isNotEmpty() && sender is Player) {
            if (Bukkit.getPlayer(args[0]) == null) {
                sender.sendMessage(Config.prefix + " §cPlayer not found!")
                return false
            }

            target = Bukkit.getPlayer(args[0]) as Player

            others = true
        }

        if (!sender.checkPermission("heal", others))
            return false

        target.health = 20.0
        target.foodLevel = 20

        when {
            others -> {
                sender.sendMessage(Config.prefix + " §a§l" + target.name + " §ahas been healed!")
                target.sendMessage(Config.prefix + " §aYou have been healed!")
            }
            else -> {
                target.sendMessage(Config.prefix + " §aYou have been healed!")
            }
        }

        return false

    }

}